base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -874,76 +878,97 @@ private X509CertImpl issueX509Cert(
-                String respS;
+                    List<LogServer> logServers= sct.getLogServerConfig();
+                    List<String> ctResponses = new ArrayList<>();
 
-                { // loop through all CTs
-                    String ct_host = "ct.googleapis.com";
-                    int ct_port = 80;
-                    String ct_uri = "http://ct.googleapis.com/testtube/ct/v1/add-pre-chain";
+                    // loop through all CT log servers
+                    for (LogServer ls : logServers) {
+                        logger.debug("Processing log server ID: " + ls.getId());
 
-                    respS = certTransSendReq(
-                            ct_host, ct_port, ct_uri, ctRequest);
+                        String ct_host = ls.getUrl().getHost();
+                        logger.debug("Log server host: " + ct_host);
 
-                    // verify the sct: TODO - not working, need to fix
-                    verifySCT(CTResponse.fromJSON(respS), cert.getTBSCertificate());
-                } // todo: should collect a list of CTResonses once out of loop
+                        int ct_port = ls.getUrl().getPort();
+                        logger.debug("Log server port: " + ct_port);
 
-                /**
-                 * Now onto turning the precert into a real cert
-                 */
-                // remove the poison extension
-                exts.delete("1.3.6.1.4.1.11129.2.4.3");
-                certi.delete(X509CertInfo.EXTENSIONS);
-                certi.set(X509CertInfo.EXTENSIONS, exts);
-
-                // create SCT extension
-                // TODO : handle multiple SCTs; should pass in list of CTResponses
-                Extension sctExt = createSCTextension(CTResponse.fromJSON(respS));
-
-                // add the SCT extension
-                exts.set(sctExt.getExtensionId().toString(), sctExt);
-                //check
-                Extension p = (Extension) exts.get("1.3.6.1.4.1.11129.2.4.2");
-                certi.delete(X509CertInfo.EXTENSIONS);
-                certi.set(X509CertInfo.EXTENSIONS, exts);
-
-                try { //double-check if it's there
-                    exts = (CertificateExtensions)
-                            certi.get(X509CertInfo.EXTENSIONS);
-                    logger.debug(method + " about to check sct ext");
-                    Extension check = (Extension) exts.get("1.3.6.1.4.1.11129.2.4.2");
-                    if ( check == null)
-                        logger.debug(method + " check not found");
-                    else
-                        logger.debug(method + " SCT ext found added successfully");
-                } catch (Exception e) {
-                    logger.debug(method + " check sct failed:" + e.toString());
+                        // TODO: Refactor to form right rest API
+                        String ct_uri = ls.getUrl() + "ct/v1/add-pre-chain";
```

Comment:
I suggest that you add an example CT log config entry in the CS.cfg complete with comments, but with ca.certTransparency.enable set to off by default.  Take a look at ca.scep as example.

Reply:
Done